### ISSUE TITLE
`fortune` does not build from source successfully

### ISSUE THREAD

**Issue by RSully** [Homebrew/homebrew-core#131516]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config

HOMEBREW_VERSION: 4.0.18
ORIGIN: https://github.com/Homebrew/brew
HEAD: b0dc84b117f54f6b18e0528b95477c54e409ebdd
Last commit: 8 days ago
Core tap JSON: 20 May 23:01 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code --new-window --wait
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_NO_ANALYTICS: set
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSECURE_REDIRECT: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.40.1 => /opt/homebrew/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: 14.3
Rosetta 2: false

$ brew doctor

Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

Build `fortune` from source

### What happened (include all command output)?

```
user@user-MacBook-Pro ~> brew install -s fortune
==> Fetching fortune
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/56b4d0fe991b70cdb48f02612e0157ca714d71b9/Formula/for
###################################################################################################################### 100.0%
==> Downloading https://www.ibiblio.org/pub/linux/games/amusements/fortune/fortune-mod-9708.tar.gz
Already downloaded: /Users/user/Library/Caches/Homebrew/downloads/6ea7ef059f5b97097d32fd3cbf05c6d8306b65fa574df67ccdc654727bd0801f--fortune-mod-9708.tar.gz
==> make install
Last 15 lines from /Users/user/Library/Logs/Homebrew/fortune/01.make:
clang -s -o fortune fortune.o 
install -m 0755 -d /opt/homebrew/Cellar/fortune/9708/bin
install -m 0755 fortune/fortune /opt/homebrew/Cellar/fortune/9708/bin
cd util && /Applications/Xcode.app/Contents/Developer/usr/bin/make CC='clang' CFLAGS='-O2 -DFORTDIR="\"/opt/homebrew/Cellar/fortune/9708/share/games/fortunes\"" -DOFFDIR="\"/opt/homebrew/Cellar/fortune/9708/share/games/fortunes/off\"" -Wall -fomit-frame-pointer -pipe'	\
		    LDFLAGS='-s'
clang -O2 -DFORTDIR="\"/opt/homebrew/Cellar/fortune/9708/share/games/fortunes\"" -DOFFDIR="\"/opt/homebrew/Cellar/fortune/9708/share/games/fortunes/off\"" -Wall -fomit-frame-pointer -pipe   -c -o strfile.o strfile.c
clang -s -o strfile strfile.o
clang -O2 -DFORTDIR="\"/opt/homebrew/Cellar/fortune/9708/share/games/fortunes\"" -DOFFDIR="\"/opt/homebrew/Cellar/fortune/9708/share/games/fortunes/off\"" -Wall -fomit-frame-pointer -pipe   -c -o unstr.o unstr.c
unstr.c:134:8: error: call to undeclared library function 'exit' with type 'void (int) __attribute__((noreturn))'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
              exit(1);
              ^
unstr.c:134:8: note: include the header <stdlib.h> or explicitly provide a declaration for 'exit'
1 error generated.
make[1]: *** [unstr.o] Error 1
make: *** [util-bin] Error 2
```

### What did you expect to happen?

Successful install

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install -s fortune
```

**Comment by SMillerDev** [comment]
Yeah, it probably needs patches to build on newer macOS versions.

**Comment by ZhongRuoyu** [comment]
This should be fixed by #131548. Please try again.
